src/binding/transaction_log_store.cpp

```diff
@@ -0,0 +1,309 @@
+
+				if (fileAgeMs <= this->retentionMs) {
+					continue; // file is too new, don't purge
+				}
+			} catch (const std::filesystem::filesystem_error& e) {
+				// file was deleted or doesn't exist
+				DEBUG_LOG("%p TransactionLogStore::purge File no longer exists: %s\n", this, logFile->path.string().c_str())
+				continue;
+			}
+		}
+
+		DEBUG_LOG("%p TransactionLogStore::purge Purging log file: %s\n", this, logFile->path.string().c_str())
+
+		// delete the log file
+		logFile->close();
+		bool removed = std::filesystem::remove(logFile->path);
+		if (visitor && removed) {
+			visitor(logFile->path);
+		}
+
+		// collect sequence number for removal
+		sequenceNumbersToRemove.push_back(entry.first);
+	}
+
+	// remove sequence files from the map
+	for (uint32_t sequenceNumber : sequenceNumbersToRemove) {
+		this->sequenceFiles.erase(sequenceNumber);
+	}
+
+	// if all log files have been removed, clean up the empty directory
+	// only try to remove if we actually removed at least one file from this store
+	if (this->sequenceFiles.empty() && !sequenceNumbersToRemove.empty()) {
+		try {
+			if (std::filesystem::exists(this->path)) {
+				std::filesystem::remove(this->path);
+				DEBUG_LOG("%p TransactionLogStore::purge Removed empty log directory: %s\n", this, this->path.string().c_str())
+			}
+		} catch (const std::filesystem::filesystem_error& e) {
+			DEBUG_LOG("%p TransactionLogStore::purge Failed to remove log directory %s: %s\n", this, this->path.string().c_str(), e.what())
+		} catch (...) {
+			DEBUG_LOG("%p TransactionLogStore::purge Unknown error removing log directory %s\n", this, this->path.string().c_str())
+		}
+	}
+}
+
+void TransactionLogStore::registerLogFile(const std::filesystem::path& path, const uint32_t sequenceNumber) {
+	std::lock_guard<std::mutex> lock(this->storeMutex);
+
+	auto logFile = std::make_unique<TransactionLogFile>(path, sequenceNumber);
+	this->sequenceFiles[sequenceNumber] = std::move(logFile);
+
+	if (sequenceNumber > this->currentSequenceNumber) {
+		this->currentSequenceNumber = sequenceNumber;
+	}
+
+	// update next sequence number to be one higher than the highest existing
+	if (sequenceNumber > this->nextSequenceNumber) {
+		this->nextSequenceNumber = sequenceNumber + 1;
+	}
+
+	DEBUG_LOG("%p TransactionLogStore::registerLogFile Added log file: %s (seq=%u)\n",
+		this, path.string().c_str(), sequenceNumber)
+}
+
+void TransactionLogStore::writeBatch(TransactionLogEntryBatch& batch) {
+	DEBUG_LOG("%p TransactionLogStore::commit Adding batch with %zu entries to store \"%s\" (timestamp=%llu)\n",
+		this, batch.entries.size(), this->name.c_str(), batch.timestamp)
+
+	std::lock_guard<std::mutex> lock(this->storeMutex);
+
+	// write entries across multiple log files until all are written
+	while (!batch.isComplete()) {
+		TransactionLogFile* logFile = nullptr;
+
+		// get the current log file and rotate if needed
+		while (logFile == nullptr && this->currentSequenceNumber) {
```

Comment:
As far as I can tell, this is not implementing https://github.com/HarperFast/rocksdb-js/pull/238/commits/6dd20de7af5f83c82ec465c5d4e375ded6691cac#diff-2872eef24b542a107a9c6f0b14427d656cf6cd3bef907f9a991b1482d71855acR18:

> A new transaction log file may be started even if the previous file has not reached its maximum size. A new transaction log file should be started if the timestamp is older than 1/8 of the transaction retention period.

(1/4 might be better though)